client/my-sites/activity/filterbar/date-range-selector.jsx

```diff
@@ -288,51 +239,44 @@ export class DateRangeSelector extends Component {
 		const { isVisible } = this.props;
 		const from = this.getFromDate();
 		const to = this.getToDate();
+		const now = new Date();
 
 		const buttonClass = classnames( {
 			filterbar__selection: true,
 			'is-selected': !! from,
 			'is-active': isVisible && ! from,
 		} );
+
 		return (
-			<Fragment>
-				<Button
-					className={ buttonClass }
-					compact
-					borderless
-					onClick={ this.handleButtonClick }
-					ref={ this.dateRangeButton }
-				>
-					{ this.getFormattedDate( from, to ) }
-				</Button>
-				{ ( from || to ) && (
-					<Button
-						className="filterbar__selection-close"
-						compact
-						borderless
-						onClick={ this.handleResetSelection }
-					>
-						<Gridicon icon="cross-small" />
-					</Button>
-				) }
-				{ isWithinBreakpoint( '>660px' ) && (
-					<Popover
-						id="filterbar__date-range"
-						isVisible={ isVisible }
-						onClose={ this.handleClose }
-						position="bottom"
-						relativePosition={ { left: -125 } }
-						context={ this.dateRangeButton.current }
-					>
-						{ this.renderDateRangeSelection() }
-					</Popover>
-				) }
-				{ ! isWithinBreakpoint( '>660px' ) && (
-					<MobileSelectPortal isVisible={ isVisible }>
-						<Card>{ this.renderDateRangeSelection() }</Card>
-					</MobileSelectPortal>
+			<DateRangePicker
```

Comment:
We should have a text label before the DatePicker which describes it's purpose. You can use a `<span>` and add a [`aria-labelledby`](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/ARIA_Techniques/Using_the_aria-labelledby_attribute) attribute to the date range to associate it with the descriptive label.

Reply:
Have you got any examples? I'm not sure I understand fully.

Reply:
As agreed we should add an issue about this and tag as a11y.

Reply:
I've popped an  issue open about this @ #31347 :)